src/core/cgroup.c

```diff
@@ -4200,7 +4200,10 @@ int unit_cgroup_freezer_action(Unit *u, FreezerAction action) {
 
         if (target == kernel) {
                 u->freezer_state = target;
-                return 0;
+                if (action == FREEZER_FREEZE)
+                        return 0;
+                else
+                        ret = 0;
```

Comment:
Previously the idea was that whenever I got reply from the systemd and then checked freezer state for given unit I'd never see "transition state" . With this change I might get reply back while thawing operation could have been started.

Reply:
Changed, see below. (That was also needed to make sure freezing won't fail later.)